pkg/logs/launchers/integration/launcher.go

```diff
@@ -85,39 +120,100 @@ func (s *Launcher) run() {
 			for _, source := range sources {
 				// TODO: integrations should only be allowed to have one IntegrationType config.
 				if source.Config.Type == config.IntegrationType {
-					logFilePath, err := s.createFile(cfg.IntegrationID)
+					logFile, err := s.createFile(cfg.IntegrationID)
 					if err != nil {
 						ddLog.Warn("Failed to create integration log file: ", err)
 						continue
 					}
-					filetypeSource := s.makeFileSource(source, logFilePath)
+					filetypeSource := s.makeFileSource(source, logFile.filename)
 					s.sources.AddSource(filetypeSource)
 
 					// file to write the incoming logs to
-					s.integrationToFile[cfg.IntegrationID] = logFilePath
+					s.integrationToFile[cfg.IntegrationID] = logFile
 				}
 			}
 
 		case log := <-s.integrationsLogsChan:
-			logFilePath := s.integrationToFile[log.IntegrationID]
+			s.receiveLogs(log)
+		case <-s.stop:
+			return
+		}
+	}
+}
 
-			err := s.ensureFileSize(logFilePath)
-			if err != nil {
-				ddLog.Warn("Failed to get file size: ", err)
-				continue
-			}
+// receiveLogs handles writing incoming logs to their respective file as well as
+// enforcing size limitations
+func (s *Launcher) receiveLogs(log integrations.IntegrationLog) {
+	fileToUpdate := s.integrationToFile[log.IntegrationID]
 
-			err = s.writeFunction(logFilePath, log.Log)
-			if err != nil {
-				ddLog.Warn("Error writing log to file: ", err)
-			}
-		case <-s.stop:
+	// Ensure the individual file doesn't exceed integrations_logs_files_max_size
+	// Add 1 because we write the \n at the end as well
+	logSize := int64(len(log.Log)) + 1
+	if fileToUpdate.size+logSize > s.fileSizeMax {
+		s.combinedUsageSize -= fileToUpdate.size
+		err := s.deleteAndRemakeFile(fileToUpdate.filename)
+		if err != nil {
+			ddLog.Warn("Failed to delete and remake oversize file", err)
 			return
 		}
+
+		fileToUpdate.size = 0
+	}
+
+	// Ensure combined logs usage doesn't exceed integrations_logs_total_usage
+	for s.combinedUsageSize+logSize > s.combinedUsageMax {
+		s.deleteLeastRecentlyModified()
+		s.updateLeastRecentlyModifiedFile()
+	}
+
+	err := s.writeLogToFileFunction(filepath.Join(s.runPath, fileToUpdate.filename), log.Log)
+	if err != nil {
+		ddLog.Warn("Error writing log to file: ", err)
+	}
+	s.combinedUsageSize += logSize
+
+	// Update information for the modified file
+	fileToUpdate.lastModified = time.Now()
+	fileToUpdate.size += logSize
+
+	// Update leastRecentlyModifiedFile
+	if s.leastRecentlyModifiedFile == fileToUpdate && len(s.integrationToFile) > 1 {
+		s.updateLeastRecentlyModifiedFile()
 	}
+
 }
 
-// writeLogToFile is used as a function pointer
+// deleteLeastRecentlyModified deletes and remakes the least recently log file
+func (s *Launcher) deleteLeastRecentlyModified() {
+	s.combinedUsageSize -= s.leastRecentlyModifiedFile.size
```

Comment:
Shome we perform this after the `err` check? If the delete failed, the combined size may not have decreased?

Reply:
Updated so usage is only updated after the error check, also changed it so this function includes error checking